src/main/scala/org/scalajs/cli/Scalajsld.scala

```diff
@@ -170,27 +213,62 @@ object Scalajsld {
 
       val linker = StandardImpl.linker(config)
       val logger = new ScalaConsoleLogger(options.logLevel)
-
-      val output = {
-        val js = options.output.toPath()
-        val sm = js.resolveSibling(js.getFileName().toString() + ".map")
-
-        def relURI(f: Path) =
-          new URI(null, null, f.getFileName().toString, null)
-
-        LinkerOutput(PathOutputFile(js))
-          .withSourceMap(PathOutputFile(sm))
-          .withSourceMapURI(relURI(sm))
-          .withJSFileURI(relURI(js))
-      }
-
       val cache = StandardImpl.irFileCache().newCache
 
       val result = PathIRContainer
         .fromClasspath(classpath)
-        .flatMap(containers => cache.cached(containers._1))
-        .flatMap(linker.link(_, moduleInitializers, output, logger))
+        .map(_._1)
+        .flatMap(cache.cached _)
+        .flatMap{irFiles =>
+          outputFile match {
+            case OutputFile.SingleFile(jsFile) =>
+              DeprecatedFileOutputSilencer().link(linker, irFiles.toList, moduleInitializers, jsFile, logger)
+            case OutputFile.Directory(outputDir) =>
+              linker.link(irFiles, moduleInitializers, PathOutputDirectory(outputDir.toPath), logger)
+          }
+        }
       Await.result(result, Duration.Inf)
     }
   }
+
+  private def createOutputFile(options: Options): Option[OutputFile] = {
+    options.output.map(OutputFile.SingleFile) orElse
+      options.outputDir.map(OutputFile.Directory)
+  }
+
+  private abstract class DeprecatedFileOutputSilencer {
```

Comment:
Add a comment? The name of the class is a bit meta IMO. (can stay, but should be commented on).

Maybe: DeprecatedLinkerAPI is a more descriptive name?